client/state/themes/test/reducer.js

```diff
@@ -402,4 +408,220 @@ describe( 'reducer', () => {
+				type: ACTIVE_THEME_REQUEST_SUCCESS,
+				siteId: 2211667,
+				themeId: 'twentysixteen',
+				themeName: 'Twentysixteen',
+				themeCost: {
+					currency: 'USD',
+					number: 0,
+					display: ''
+				}
+			} );
+
+			expect( state ).to.have.keys( [ '2211667' ] );
+			expect( state ).to.eql( { 2211667: 'twentysixteen' } );
+		} );
+
+		it( 'should track theme activate request success', () => {
+			const state = activeThemes( deepFreeze( {} ), {
+				meta: {
+					analytics: [
+						{
+							payload: {
+								name: 'calypso_themeshowcase_theme_activate',
+								properties: {
+									previous_theme: 'twentyfifteen',
+									purchased: false,
+									search_term: 'simple, white',
+									source: 'unknown',
+									theme: 'twentysixteen',
+								},
+								service: 'tracks',
+							},
+							type: 'ANALYTICS_EVENT_RECORD'
+						},
+					],
+				},
+				type: THEME_ACTIVATE_REQUEST_SUCCESS,
+				theme: { id: 'twentysixteen' },
+				siteId: 2211888,
+			} );
+
+			expect( state ).to.have.keys( [ '2211888' ] );
+			expect( state ).to.eql( { 2211888: 'twentysixteen' } );
+		} );
+
+		it( 'should persist state', () => {
+			const original = deepFreeze( activeThemes( deepFreeze( {} ), {
+				meta: {
+					analytics: [
+						{
+							payload: {
+								name: 'calypso_themeshowcase_theme_activate',
+								properties: {
+									previous_theme: 'twentyfifteen',
+									purchased: false,
+									search_term: 'simple, white',
+									source: 'unknown',
+									theme: 'twentysixteen',
+								},
+								service: 'tracks',
+							},
+							type: 'ANALYTICS_EVENT_RECORD'
+						},
+					],
+				},
+				type: THEME_ACTIVATE_REQUEST_SUCCESS,
+				theme: { id: 'twentysixteen' },
+				siteId: 2211888,
+			} ) );
+
+			const state = activeThemes( original, { type: SERIALIZE } );
+
+			expect( state ).to.eql( { 2211888: 'twentysixteen' } );
+		} );
+
+		it( 'should load valid persisted state', () => {
+			const original = deepFreeze( { 2211888: 'twentysixteen' } );
+
+			const state = activeThemes( original, { type: DESERIALIZE } );
+			expect( state ).to.eql( { 2211888: 'twentysixteen' } );
+		} );
+
+		it( 'should not load invalid persisted state', () => {
+			const original = deepFreeze( {
+				2916284: 1234
+			} );
+
+			const state = activeThemes( original, { type: DESERIALIZE } );
+			expect( state ).to.eql( {} );
+		} );
+	} );
+
+	describe( 'activeThemeRequest', () => {
+		it( 'should default to an empty object', () => {
+			const state = activeThemeRequest( undefined, {} );
+			expect( state ).to.eql( {} );
+		} );
+
+		it( 'should map site ID to true value if request in progress', () => {
+			const state = activeThemeRequest( deepFreeze( {} ), {
+				type: ACTIVE_THEME_REQUEST,
+				siteId: 2916284,
+			} );
+
+			expect( state ).to.eql( {
+				2916284: true
+			} );
+		} );
+
+		it( 'should accumulate mappings', () => {
+			const state = activeThemeRequest( deepFreeze( {
```

Comment:
Indentation is a bit weird in this test's `state` creation, and in the following ones. Should probably be

```js
const state = activeThemeRequest(
	deepFreeze( { 2916284: true } ),
	{
		/*...*/
	}
);
```

as that makes clearer what the initial state is, and which is the action.